### PR TITLE
feat(notifier): add new block event

### PR DESCRIPTION
**Pull request by vjgene**
Adds new block event functionality to add events page - RGWY690

### CODE REVIEW

**Review by itofarina — COMMENTED** [review]
I'm not sure I understood the goal of this PR. Is it rebased on https://github.com/rsksmart/rif-marketplace-ui/pull/849 and should we merge that one before this one?

File: src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx [hunk 1]
```diff
@@ -71,13 +72,20 @@ const NotifierOffersSelectedPage: FC = () => {
 
   const [events, setEvents] = useState<Array<EventItem>>([])
   const [addEventCollapsed, setAddEventCollapsed] = useState<boolean>(false)
+  const [key, setKey] = useState<number>(0)
```
Comment (jurajpiar) [inline src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx]: don't store DOM keys in a state like this.

File: src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx [hunk 2]
```diff
@@ -71,13 +72,20 @@ const NotifierOffersSelectedPage: FC = () => {
...
 
   const addNotifierEvent = (notifierEvent: NotifierEvent): void => {
+    if (notifierEvent.type === SUPPORTED_EVENTS.NEWBLOCK
+        && events.find((addedEvent) => addedEvent.type === SUPPORTED_EVENTS.NEWBLOCK)) {
```
Comment (jurajpiar) [inline src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx]: the repetition of `notifierEvent.type === SUPPORTED_EVENTS.NEWBLOCK` seems to warrant refactor to a function.

File: src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx [hunk 3]
```diff
@@ -124,6 +132,7 @@ const NotifierOffersSelectedPage: FC = () => {
       <Button
         onClick={() => {
           setAddEventCollapsed(!addEventCollapsed)
+          setKey(key + 1)
```
Comment (jurajpiar) [inline src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx:135]: delete

File: src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx [hunk 4]
```diff
@@ -135,7 +144,11 @@ const NotifierOffersSelectedPage: FC = () => {
       <br />
       <Collapse in={addEventCollapsed}>
         <Box mt={6}>
-          <NotificationEventCreate onAddEvent={addNotifierEvent} channels={order?.item.channels} />
+          <NotificationEventCreate
+            key={key}
```
Comment (jurajpiar) [inline src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx]: using indexes for keys is discouraged. if you don't have anything that would uniquely identify an event, then convert it to string.

**Review by jurajpiar — CHANGES_REQUESTED** [review]
Good good, just don't use react state to store dom keys.

File: src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx (hunk 1 above)
Comment (vjgene) [reply src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx]: using generator function instead for keys

File: src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx (hunk 2 above)
Comment (vjgene) [reply src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx]: done

File: src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx (hunk 4 above)
Comment (vjgene) [reply src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx]: using generator function

File: src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx [hunk 5]
```diff
@@ -98,6 +109,8 @@ const NotifierOffersSelectedPage: FC = () => {
     actions: <RemoveButton id={event.id} handleSelect={removeEvent} />,
   }))
 
+  const generateKey = (): string => `'eve_'${new Date().getTime()}`
```
Comment (jurajpiar) [inline src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx:112]: I recommend the following react docs to understand what to use for keys: https://reactjs.org/docs/lists-and-keys.html. The rule of thumb is: use unique and static values for keys.

**Review by jurajpiar — CHANGES_REQUESTED** [review]

File: src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx [hunk 6]
```diff
@@ -135,7 +148,11 @@ const NotifierOffersSelectedPage: FC = () => {
       <br />
       <Collapse in={addEventCollapsed}>
         <Box mt={6}>
-          <NotificationEventCreate onAddEvent={addNotifierEvent} channels={order?.item.channels} />
+          <NotificationEventCreate
+            key={generateKey()}
```
Comment (itofarina) [inline src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx:152]: Do we need a key here?

**Review by itofarina — CHANGES_REQUESTED** [review]
I might be missing something, but I didn't get the use of a `key` prop on a single component, is it supposed to be under some iteration?
Apart from that it looks great!

File: src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx (hunk 5 above)
Comment (vjgene) [reply src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx:112]: using key to reinitailize component.. reference here - https://kentcdodds.com/blog/understanding-reacts-key-prop

File: src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx (hunk 6 above)
Comment (vjgene) [reply src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx:152]: using key to reinitailize component.. reference here - https://kentcdodds.com/blog/understanding-reacts-key-prop

File: src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx (hunk 5 above)
Comment (itofarina) [inline src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx:112]:
```suggestion
  const generateKey = (): string => `eve_${new Date().getTime()}`
```

File: src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx (hunk 6 above)
Comment (itofarina) [reply src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx:152]: Ah ok, I didn't think about using it this way. Thanks for the reference 👍.

**Review by itofarina — APPROVED** [review]

File: src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx (hunk 6 above)
Comment (jurajpiar) [reply src/components/pages/notifier/buy/NotifierOffersSelectedPage.tsx:152]: Well whilst the article is certainly enlightening I personally still don't understand how exactly is your code benefiting from using this reinitialisation. Could you elaborate on that please?

**Review by jurajpiar — APPROVED** [review]
Nice one

**Pull request merged**